### ISSUE TITLE
is there any way to refresh or restart the server ???

### ISSUE THREAD

**Issue by mushfiq-cse05-buet** [esp8266/Arduino#2271]
whenever i run a server such as LED on/off, after uncertain amount of time the server get lost or inaccessible from my browser, so i'm looking for a way to refresh or restart the server again ?? TIA

**Comment by krzychb** [comment]
Does https://github.com/esp8266/Arduino/issues/2272#issuecomment-232256697 aswer your question or this is another issue?

**Comment by mushfiq-cse05-buet** [comment]
@krzychb  no its an another issue i wanto refresh the server after a certain interval of time

**Comment by mushfiq-cse05-buet** [comment]
@krzychb 
the reason behind this refreshing is , whenever i run a simple LED ON/OFF server on my esp-01, it runs smoothly for 2-3 hours, but after this interval of time the server get lost or not accessible from my browser! restarting the module by ESP.restart() is a solution but as it restart the whole module and the LED is connected to gpio, the last state of the LED also get lost! so i think i need to refresh the server after a certain interval to keep state of connected LED unchanged  TIA

**Comment by krzychb** [comment]
@mushfiq-cse05-buet,

You can save state of LED in [EEPROM](https://github.com/esp8266/Arduino/blob/master/doc/libraries.md#eeprom) before executing ESP.restart().

However, I believe there is something fundamentally wrong with module, server's application or network if server gets lost or is not accessible from browser after 2-3 hours. I would connect esp-01 running server to a serial terminal, enable additinal [debugging](https://github.com/esp8266/Arduino/blob/master/doc/Troubleshooting/debugging.md) if required and look for clues why connection is lost. 

I have been [testing](https://github.com/krzychb/EspFire2012) several esp modules that run webserver together with led strip control. If application is compatible, configured and tested for ESP, I did not face any communication / performance issues for several days (I was not patient enough to run them for longer time).

**Comment by mushfiq-cse05-buet** [comment]
@krzychb thanks fro ur reply, but is there any function which can refresh or restart the server object ???

**Comment by rima888** [comment]
Hi, I have the same problem. Do you found a solution to the problem? I'm appreciate if you shared to as.

**Comment by vvrona** [comment]
Siemka krzychb i have that same problem that you have, i have relay to switch light on my desk but after couple ours esp is not respond. i change a esp board for new one and now it work one day, and don`t response again. Do you solved this ?

**Comment by d-a-v** [comment]
Could you please try PR #3362 and report back ?
Could you please share a simple sketch that would allow us to reproduce the problem ?
Thanks

**Comment by devyte** [comment]
@mushfiq-cse05-buet waiting on feedback.

**Comment by devyte** [comment]
Closing due to no feedback in over 2 months. I think this issue is the same as, or related to, #2330 .
